gitolize.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+
+# A script wrapping a shell command so that it runs with a directory populated from a git repository.
+# In the write mode, the repository is locked with a temporary file and
+# changes made in that directory by the shell command get pushed back to the repository.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+  echo "Usage: $0 [-b <branch>] [-d <git_directory>] [-l <local_directory>] [-m message] [-w] [command...]" 1>&2
+  exit 1
+}
+
+git() {
+  COMMAND="$1"
+  shift
+  EXTRA_FLAGS=
+  case "$COMMAND" in
+    add)
+      EXTRA_FLAGS="$GIT_VERBOSE_FLAG"
+      ;;
+    *)
+      EXTRA_FLAGS="$GIT_QUIET_FLAG"
+      ;;
+  esac
+  if [[ "$VERBOSE" ]]
+  then
+    echo "running $GIT -C "$LOCAL_DIRECTORY" $COMMAND $EXTRA_FLAGS $@" 1>&2
+  fi
+  $GIT -C "$LOCAL_DIRECTORY" $COMMAND $EXTRA_FLAGS "$@"
+}
+
+LOCK_FILE=gitolize.lock
+
+EXIT_CODE=0
+
+GIT="$(which git)"
+GIT_QUIET_FLAG=--quiet
+GIT_VERBOSE_FLAG=
+
+BRANCH=main
+GIT_DIRECTORY=.
+LOCAL_DIRECTORY=
+CLEANUP_LOCAL_DIRECTORY=
+COMMIT_MESSAGE=
+VERBOSE=
+WRITE=
+
+
+while getopts b:d:l:m:vw OPTION
+do
+  case $OPTION in
+    b)
+      BRANCH="$OPTARG"
+      ;;
+    d)
```

Comment:
We could also consider another letter like `-p` (project), as it doesn't have to be about a directory.
The purpose is to allow multiple workflows to work on separate sets of files without locking each other.

Reply:
Done in 64454d9d8dc59a98c9dc1c45bc65e2cc6d283fcf .